api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/BreastfeedingAndPregnantCalculation.java

```diff
@@ -0,0 +1,267 @@
+			List<Obs> markedPregnantByWeeksList = new ArrayList<>();
+			if (markedPregnantByWeeksObsResult != null && !markedPregnantByWeeksObsResult.isEmpty()) {
+				markedPregnantByWeeksList = EptsCalculationUtils.extractResultValues(markedPregnantByWeeksObsResult);
+			}
+			
+			ListResult markedPregnantOnEddObsResult = (ListResult) markedPregnantOnEdd.get(pId);
+			List<Obs> markedPregnantOnEddList = new ArrayList<>();
+			if (markedPregnantOnEddObsResult != null && !markedPregnantOnEddObsResult.isEmpty()) {
+				markedPregnantOnEddList = EptsCalculationUtils.extractResultValues(markedPregnantOnEddObsResult);
+			}
+			
+			// evaluating breastfeeding maps
+			ListResult deliveryDateMapObsResult = (ListResult) deliveryDateMap.get(pId);
+			List<Obs> deliveryDateMapList = new ArrayList<>();
+			if (deliveryDateMapObsResult != null && !deliveryDateMapObsResult.isEmpty()) {
+				deliveryDateMapList = EptsCalculationUtils.extractResultValues(deliveryDateMapObsResult);
+			}
+			
+			ListResult criteriaHivStartMapObsResult = (ListResult) criteriaHivStartMap.get(pId);
+			List<Obs> criteriaHivStartMapList = new ArrayList<>();
+			if (criteriaHivStartMapObsResult != null && !criteriaHivStartMapObsResult.isEmpty()) {
+				criteriaHivStartMapList = EptsCalculationUtils.extractResultValues(criteriaHivStartMapObsResult);
+			}
+			
+			ListResult lactatingMapObsResult = (ListResult) lactatingMap.get(pId);
+			List<Obs> lactatingMapList = new ArrayList<>();
+			if (lactatingMapObsResult != null && !lactatingMapObsResult.isEmpty()) {
+				lactatingMapList = EptsCalculationUtils.extractResultValues(lactatingMapObsResult);
+			}
+			
+			PatientProgram patientProgram = null;
+			PatientState breastfeedingState = null;
+			PatientState pregnantState = null;
+			
+			if (female.contains(pId) && lastVlObs != null && lastVlObs.getObsDatetime() != null && criteria != null
+			        && onArtForMoreThan3Months.contains(pId)) {
+				Date pregnancyStartDate = EptsCalculationUtils.addMonths(lastVlObs.getObsDatetime(), -9);
+				Date breastfeedingStartDate = EptsCalculationUtils.addMonths(lastVlObs.getObsDatetime(), -18);
+				Date endDate = lastVlObs.getObsDatetime();
+				
+				if (result != null) {
+					patientProgram = (PatientProgram) result.getValue();
+				}
+				// get patient_state for pregnant and breastfeeding
+				if (patientProgram != null) {
+					for (PatientState patientState : patientProgram.getCurrentStates()) {
+						if (patientState.getState().getProgramWorkflowStateId() == 27) {
+							pregnantState = patientState;
+							
+						}
+						if (patientState.getState().getProgramWorkflowStateId() == 25) {
+							breastfeedingState = patientState;
+						}
+					}
+				}
+				
+				if (criteria.equals(BreastfeedingAndPregnant.PREGNANT)) {
+					if (markedPregnantList.size() > 0) {
+						for (Obs obs : markedPregnantList) {
+							if (obs != null && obs.getObsDatetime() != null
+							        && (obs.getObsDatetime().equals(pregnancyStartDate)
```

Comment:
I suggest we use `obs.getObsDatetime().compareTo(pregnancyStartDate) >= 0`. That way the if statement will have only one expression and be a bit more readable.

Reply:
Done